velox/connectors/Connector.h

```diff
@@ -287,6 +287,67 @@ class DataSource {
   }
 };
 
+/// Represents a lookup request for a given input.
+struct LookupRequest {
+  /// Contains the input column vectors used by lookup join and range
+  /// conditions.
+  RowVectorPtr input{nullptr};
+
+  LookupRequest(RowVectorPtr input) : input(std::move(input)) {}
+};
+
+/// Represents the lookup result for a subset of input produced by the
+/// 'LookupResultIterator'.
+struct LookupResult {
```

Comment:
Would it make sense to move these structs inside IndexSource class? Alternatively, we may need to extend the names to add 'index': IndexLookupRequest/Result